Add button to reset processing config

helenst opened the pull request:
https://jiscdev.atlassian.net/browse/RDSSARK-431 - builtin processing configs get a "Reset" button on the Processing config page, so they can be reset to original values.

See also https://github.com/JiscRDSS/rdss-archivematica/pull/186

sevein committed:
Dashboard: add MCP config download button

helenst committed:
Add button to reset processing config

sevein commented on src/dashboard/src/templates/administration/processing.html:
```diff
@@ -25,6 +25,9 @@ <h3>{% trans "Processing configuration" %}</h3>
           <td>
             <a class="btn btn-default btn-sm" href="{% url "components.administration.views_processing.edit" item.1 %}">{% trans "Edit" %}</a>
+            {% if item.1 == "default" or item.1 == 'automated' %}
+            <a class="btn btn-default btn-sm" title="Reset to original values" href="{% url "components.administration.views_processing.reset" item.1 %}">{% trans "Reset" %}</a>
```
Nitpick: this may not be very idiomatic but notice how we indent code inside template blocks.

sevein commented on src/dashboard/src/components/administration/views_processing.py:
```diff
@@ -26,6 +26,7 @@
 from django.http import Http404
 
 from components import helpers
+from processing import install_builtin_config
```
How's this going to work when you run Dashboard and MCPServer in separate containers? If you take a look at `dashboard.Dockerfile` you'll notice that the sources of MCPserver are not made available.

If you want to make a function in MCPServer available over the network then you can use our poor Gearman-based RPC interface (soon to become a gRPC-based proper interface). See [RPCserver](https://github.com/artefactual/archivematica/blob/f48521e38ba38a85cec3e07aec9865737d25bbc9/src/MCPServer/lib/RPCServer.py) in https://github.com/artefactual/archivematica/pull/936. Adding new methods to the RPCServer handler is fairly easy.

helenst replied:
The `processing` module is part of archivematicaCommon so should be present in both containers... unless I've misunderstood something there. Although depending on what's best for the deployment situation, it might be good to have that function available over the network anyway.

sevein reviewed (approved):
I think it's a nice idea to have a new processing config.

I've added some comments.

sevein reviewed (changes requested):
Wrong button, sorry.

helenst committed:
Allow processing config to be regenerated via API

* support DELETE to remove a config
* GET will regenerate the config file if not present

helenst commented:
I've addressed comments and added an API endpoint to match (including a DELETE to trigger recreation)

I haven't got as far as the xml doc comparison and only bringing in new choice configurations, as detailed here: https://docs.google.com/document/d/1lQG6tTLzWaScEqdHTg38Ccxkz9zMjefgTF8mZjrKPt4/edit# - it's been taking longer than expected and I unfortunately had some hardware failures last week which didn't help matters. So at the moment GET will return existing config, or add from default if it doesn't exist. There are tests too.

I can continue with that but this may also stand well enough on its own to do for now.

sevein committed:
Dashboard: hide proc-cfg delete button for automated

Both `default` and `automated` are protected.

sevein committed:
RDSS: update processing configurations

These are the new defauls that we want to include for RDSS users.
